In [19]:
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
from mpl_toolkits.mplot3d import Axes3D

In [20]:
comisiones = [22,13,33,38,23,47,29,38,41,32,20,13,47,
              38,44,29,38,37,14,34,25,27,25,43,34]

llamadas = [139,132,144,142,142,142,138,139,144,134,135,137,146,
            146,144,147,144,149,131,144,132,132,127,154,147]

millas = [2371,2226,2731,3351,2289,3449,3114,3342,2842,2625,2121,2219,3463,
          3290,3103,2122,2791,3209,2287,2848,2690,2933,2671,2988,2829]

In [21]:
df = pd.DataFrame({
    "Comisiones": comisiones,
    "Llamadas": llamadas,
    "Millas": millas
})

In [22]:
fig = plt.figure()
ax = fig.add_subplot(111, projection='3d')

ax.scatter(df["Llamadas"], df["Millas"], df["Comisiones"])
ax.set_xlabel("Llamadas")
ax.set_ylabel("Millas")
ax.set_zlabel("Comisiones")
ax.set_title("Correlación múltiple")

plt.show()

In [23]:
corr_llamadas = np.corrcoef(llamadas, comisiones)[0,1]
corr_millas = np.corrcoef(millas, comisiones)[0,1]

corr_llamadas, corr_millas

(np.float64(0.6797156955013891), np.float64(0.8322714071792503))

La gráfica de correlación múltiple muestra la relación entre las comisiones obtenidas por los agentes de ventas y dos variables explicativas: el número de llamadas de ventas y las millas recorridas. Se observa que los puntos siguen una tendencia ascendente, indicando que mayores valores de llamadas y millas se asocian con mayores comisiones.

El análisis visual sugiere una relación positiva entre las comisiones y ambas variables independientes. La dispersión de los puntos indica que la relación es más fuerte con las millas recorridas que con el número de llamadas, lo que sugiere que la actividad de desplazamiento de los agentes tiene mayor impacto en las comisiones obtenidas.

Los coeficientes de correlación de Pearson obtenidos fueron:

Comisiones y llamadas: r =  0.68
Comisiones y millas recorridas: r = 0.83

Estos valores confirman que existe una correlación positiva moderada-alta entre las comisiones y el número de llamadas, y una correlación positiva fuerte entre las comisiones y las millas recorridas, validando el análisis realizado a partir de la gráfica.